### DISCUSSION TITLE
Cannot read property 'signals' of undefined

### DISCUSSION THREAD

**Discussion by pranitmahajan** [cypress-io/cypress#17101] [General]
I get this error after updating mac os to latest version now. Anyone else getting this error?

```
TypeError
The following error originated from your test code, not from Cypress.

  > Cannot read property 'signals' of undefined

When Cypress detects uncaught errors originating from your test code it will automatically fail the current test.

Cypress could not associate this error to any specific test.

We dynamically generated a new test to display this failure.
node_modules/human-signals/build/src/signals.js:30:1
  28 | const{
  29 | signals:{[name]:constantSignal}}=
> 30 | _os.constants;
     | ^
  31 | const supported=constantSignal!==undefined;
  32 | const number=supported?constantSignal:defaultNumber;
  33 | return{name,number,description,supported,action,forced,standard
```

**Comment by jennifer-shehane** [comment]
- Can you provide the Mac OS version? 
- What types of tests are you writing in Cypress? Component or e2e?
- Can you recreate this issue when running on a standard example project in Cypress? Or is it specific to your project?

**Reply by pranitmahajan** [reply]
Thanks for reply, below are the details:

Mac OS version is 11.4
Tests are e2e tests
Also, I am using cucumber plugin as tests are written in BDD format.

All tests were running fine till day before I upgraded my Mac OS.

I tried to dig bit into issue and my observations are:
In cypress there is dependency on execa and execa in turn as dependency on human-signals npm. This issue is raised from human-signals when starting the run.

Actually execa is using very old version of human-signals. If we upgrade it to latest that might resolve issue.

**Reply by SomvanshiGit** [reply]
How to upgrade it @pranitmahajan . can you please guide ? waiting for your reply . big thanks in advance

**Comment by rohank17** [comment]
I'm also facing the same issue, if I make some changes it's start throwing same above error.

**Comment by ravinder58** [comment]
It might be late reply but to resolve this issue, an extra line of code was auto generated which I commented out and it was running fine then,

//import cypress from 'cypress'

**Reply by midays** [reply]
Hi, I'm having the same issue, could you please tell me in which file this line is generated ?

**Reply by SomvanshiGit** [reply]
On which file //import cypress from 'cypress' you have removed the code ? @ravinder58 
please help

**Reply by ravinder58** [reply]
@midays @SomvanshiGit , it was getting generated in the spec.js files on top, where there is begining of test case. You can try removing it and it should work.

**Reply by RahulDharam2194** [reply]
@ravinder58 , Hi, I am facing the same issue and till yesterday all the scripts were running fine. I didn't changed anything. Can you please help me on this. 
Note: I have not added //import cypress From 'cypress' line in my .js file. 

Cypress version: V12.17.0

**Reply by sridharanvaishna** [reply]
@RahulDharam2194 was your issue resolved? I am also facing the same issue and I don't have //import cypress From 'cypress' line in my .js file.

**Reply by sridharanvaishna** [reply]
@ravinder58 is this issue resolved for you, I am not having //import cypress From 'cypress' line in my .js file but I m still facing this issue

**Reply by neethuvs** [reply]
@ravinder58 I have resolved the issue by removing the import statement from the command.js file and now it's working. Thank You.

**Comment by Yerriswamy23** [comment]
Hi, i am facing the same issue on windows and i don't have import cypress code in my spec . Please guide me how to resolve this issue

**Reply by ravinder58** [reply]
Can you check you test files (spec.js/cy.js), if you have any unnecessary imports getting generated. Remove them and retry againa 
@Yerriswamy23

**Reply by Yerriswamy23** [reply]
Hi @ravinder58 
It is not causing for any particular spec .it is happening for all the spec files I am unable to run any spec.

**Reply by syphax-o** [reply]
i faced the same issue i just removed import cypress from 'cypress'; from my commands.js file

**Reply by mvermadev** [reply]
Indeed - its solving the issue for me too. Thanks for sharing :)

**Comment by kasra6** [comment]
I have the same issue
it's not getting resolved

no auto added 'import' line has been added
Could you please provide advice how to solve it?

Win 11

**Reply by kasra6** [reply]
It solved when I removed 
const cypress = require("cypress")

from commands.js file

so these kind of issues usually happen when these 'import...' or 'const....' lines are automatically added to the file

if one test is not running check out the first lines of that specific file
if all tests are not working check out the code of files, that all test cases are using for example: 
support-> commands.js
support-> e2e.js
config.js
etc
